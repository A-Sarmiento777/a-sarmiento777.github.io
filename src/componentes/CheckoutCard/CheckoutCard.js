import React, { useEffect } from 'react';
import { useHistory } from 'react-router';

const CheckoutCard = ({ removeItemFromCart, increment, decrement, getCart, cart, producto }) => {
    const history = useHistory()
    let amount = cart && cart?.length && cart.map(a => parseInt(a.price) * parseInt(a.quantity))
    let totalAmount = amount && amount.length && amount?.reduce((a, b) => a + b, 0)
    useEffect(() => {
        getCart()
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])
    return (
        <tr>
            <td><img src={process.env.PUBLIC_URL + `/Assets/${producto?.type}/${producto?.name}.jpg`} width='100' alt="logos" className="img-fluid py-2" /></td>
            <td>{producto?.name}</td>
            <td><button onClick={() => decrement(producto?.id, producto?.quantity)}>Decrease</button>{'  '}{producto?.quantity}{'  '}<button onClick={() => increment(producto?.id, producto?.quantity)}>Increase</button></td>
            <td>{producto?.price * producto?.quantity}</td>
            <td onClick={() => removeItemFromCart(producto?.id)} style={{ color: 'red' }}>Remove</td>
        </tr>
    );
}


export default CheckoutCard;