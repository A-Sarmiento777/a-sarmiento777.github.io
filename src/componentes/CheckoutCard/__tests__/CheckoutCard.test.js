// Link.react.test.js
import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import CheckoutCard from "../CheckoutCard";

test("CheckoutCard Card test", () => {
  const removeItemFromCart = jest.fn(() => 1);

  let testRenderer = renderer.create(
    <CheckoutCard
    removeItemFromCart={removeItemFromCart}
    increment={() => {}}
    decrement={() => {}}
    getCart={() => {}}
    cart={[]}
    producto={{id: 1, name: "Test", quantity: 1, type: 'asd', price: "6.7"}}
    ></CheckoutCard>
  );

  let tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[4].props.onClick();

  expect(removeItemFromCart).toHaveBeenCalled();
  expect(removeItemFromCart).toBeCalledTimes(1);

  testRenderer.update(
    <Router>
      <CheckoutCard
        removeItemFromCart={removeItemFromCart}
        increment={() => {}}
        decrement={() => {}}
        getCart={() => {}}
        cart={['Testting']}
        producto={{id: 1, name: "Test", quantity: 1, type: 'asd', price: "6.7"}}
      ></CheckoutCard>
    </Router>
  );

  tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});
