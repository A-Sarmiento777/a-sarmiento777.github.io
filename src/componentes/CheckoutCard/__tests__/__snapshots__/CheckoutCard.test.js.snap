// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutCard Card test 1`] = `
<tr>
  <td>
    <img
      alt="logos"
      className="img-fluid py-2"
      src="/Assets/asd/Test.jpg"
      width="100"
    />
  </td>
  <td>
    Test
  </td>
  <td>
    <button
      onClick={[Function]}
    >
      Decrease
    </button>
      
    1
      
    <button
      onClick={[Function]}
    >
      Increase
    </button>
  </td>
  <td />
  <td
    onClick={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    Remove
  </td>
</tr>
`;

exports[`CheckoutCard Card test 2`] = `
<tr>
  <td>
    <img
      alt="logos"
      className="img-fluid py-2"
      src="/Assets/asd/Test.jpg"
      width="100"
    />
  </td>
  <td>
    Test
  </td>
  <td>
    <button
      onClick={[Function]}
    >
      Decrease
    </button>
      
    1
      
    <button
      onClick={[Function]}
    >
      Increase
    </button>
  </td>
  <td />
  <td
    onClick={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    Remove
  </td>
</tr>
`;
